plugins/glustershd/api/resp.go

```diff
@@ -15,11 +16,11 @@ type BrickHealInfo struct {
 	HostID                 string     `xml:"hostUuid,attr" json:"host-id"`
 	Name                   string     `xml:"name" json:"name"`
 	Status                 string     `xml:"status" json:"status"`
-	TotalEntries           *int       `xml:"totalNumberOfEntries" json:"total-entries,omitempty"`
-	EntriesInHealPending   *int       `xml:"numberOfEntriesInHealPending" json:"entries-in-heal-pending,omitempty"`
-	EntriesInSplitBrain    *int       `xml:"numberOfEntriesInSplitBrain" json:"entries-in-split-brain,omitempty"`
-	EntriesPossiblyHealing *int       `xml:"numberOfEntriesPossiblyHealing" json:"entries-possibly-healing,omitempty"`
-	Entries                *int       `xml:"numberOfEntries" json:"entries,omitempty"`
+	TotalEntries           *string    `xml:"totalNumberOfEntries" json:"total-entries,omitempty"`
```

Comment:
why these are converted into a string? can you update the commit message with more info

Reply:
keeping it `int` may be good in the response. Use `string` while parsing xml output and add validation to check `-`

```
TotalEntriesRaw *string `xml:"totalNumberOfEntries" json:"-"`
TotalEntries *int `json:"total-entries,omitempty"`
```

and update,

```
if healinfo.TotalEntriesRaw == "-" {
    healinfo.TotalEntries = -1
} else {
    healinfo.TotalEntries = parseint(healinfo.TotalEntriesRaw)
}
```

Reply:
@aravindavk  But we are not using these entries in any kind of mathematical operations, these are used only while displaying heal info, so I am not sure if adding so many if conditions just to handle this is okay. Its not just TotalEntries there are other fields that I will need to handle which will increase the lines of code and can be avoided here.